src/frontends/onnx/frontend/src/op/shape.cpp

```diff
@@ -4,14 +4,45 @@
 
 #include "op/shape.hpp"
 
+#include <cstdint>
+
+#include "core/null_node.hpp"
+#include "openvino/core/node_vector.hpp"
+#include "openvino/core/type/element_type.hpp"
 #include "openvino/op/shape_of.hpp"
+#include "openvino/op/slice.hpp"
 
 using namespace ov::op;
 
 namespace ov {
 namespace frontend {
 namespace onnx {
 namespace op {
+
+namespace set_15 {
+
+ov::OutputVector shape(const ov::frontend::onnx::Node& node) {
+    using ov::op::util::is_null;
+
+    const auto data = node.get_ov_inputs().at(0);
+    auto input_shape = std::make_shared<v3::ShapeOf>(data);
+
+    auto start_val = node.get_attribute_value<int64_t>("start", 0);
+    auto end_val = node.get_attribute_value<int64_t>("end", INT64_MAX);
```

Comment:
Detail, actually it's worth to use `const` where possible. (Applicable to all of the variables).
```suggestion
    const auto input_shape = std::make_shared<v3::ShapeOf>(data);

    const auto start_val = node.get_attribute_value<int64_t>("start", 0);
    const auto end_val = node.get_attribute_value<int64_t>("end", INT64_MAX);
```